### PR TITLE
fix(sdk): Name is now a mandatory param to all connectors

### PR DESCRIPTION
**Pull request by pimeys**
# Description

The SDK side of https://github.com/grafbase/api/pull/2678

# Type of change

- [ ] 💔 Breaking
- [ ] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

**Review by obmarg — APPROVED** [review]

**Comment by notrab** [comment]
@pimeys it's worth updating the README me also in this PR. We could also consider removing the README contents and link to our own documentation if you prefer to not manage multiple READMEs.

File: packages/grafbase-sdk/tests/unit/graphql.test.ts [hunk 1]
```diff
@@ -65,11 +67,11 @@ describe('GraphQL connector', () => {
   })
 
   it('combines multiple apis into one extension', () => {
-    const contentful = connector.GraphQL({
+    const contentful = connector.GraphQL('Contentful', {
       url: 'https://graphql.contentful.com/content/v1/spaces/{{ env.CONTENTFUL_SPACE_ID }}/environments/{{ env.CONTENTFUL_ENVIRONMENT }}'
     })
 
-    const github = connector.GraphQL({
+    const github = connector.GraphQL('GitHug', {
```
Comment (notrab) [inline packages/grafbase-sdk/tests/unit/graphql.test.ts:74]:
```suggestion
    const github = connector.GraphQL('GitHub', {
```

Intentional?

File: packages/grafbase-sdk/tests/unit/graphql.test.ts [hunk 2]
```diff
@@ -79,10 +81,12 @@ describe('GraphQL connector', () => {
     expect(renderGraphQL(config({ schema: g }))).toMatchInlineSnapshot(`
       "extend schema
         @graphql(
+          name: "Contentful"
           namespace: "Contentful"
           url: "https://graphql.contentful.com/content/v1/spaces/{{ env.CONTENTFUL_SPACE_ID }}/environments/{{ env.CONTENTFUL_ENVIRONMENT }}"
         )
         @graphql(
+          name: "GitHug"
```
Comment (notrab) [inline packages/grafbase-sdk/tests/unit/graphql.test.ts:89]:
```suggestion
          name: "GitHub"
```

File: packages/grafbase-sdk/tests/unit/openapi.test.ts [hunk 3]
```diff
@@ -16,13 +16,14 @@ describe('OpenAPI generator', () => {
...
         @openapi(
+          name: "Stripe"
           schema: "https://raw.githubusercontent.com/stripe/openapi/master/openapi/spec3.json"
         )"
     `)
   })
 
   it('generates the maximum possible OpenAPI datasource', () => {
-    const stripe = connector.OpenAPI({
+    const stripe = connector.OpenAPI('Strike', {
```
Comment (notrab) [inline packages/grafbase-sdk/tests/unit/openapi.test.ts:26]:
```suggestion
    const stripe = connector.OpenAPI('Stripe', {
```

Intentional?

File: packages/grafbase-sdk/tests/unit/openapi.test.ts [hunk 4]
```diff
@@ -47,6 +48,7 @@ describe('OpenAPI generator', () => {
     expect(renderGraphQL(config({ schema: g }))).toMatchInlineSnapshot(`
       "extend schema
         @openapi(
+          name: "Strike"
```
Comment (notrab) [inline packages/grafbase-sdk/tests/unit/openapi.test.ts:51]:
```suggestion
          name: "Stripe"
```

File: packages/grafbase-sdk/tests/unit/graphql.test.ts (hunk 1 above)
Comment (pimeys) [reply packages/grafbase-sdk/tests/unit/graphql.test.ts:74]: Intentional. I want the string to be different from the namespace here.

File: packages/grafbase-sdk/tests/unit/openapi.test.ts (hunk 4 above)
Comment (pimeys) [reply packages/grafbase-sdk/tests/unit/openapi.test.ts:51]: Intentional.

**Review by obmarg — APPROVED** [review]

**Pull request merged**